apps/emqx_bridge_gcp_pubsub/src/emqx_bridge_gcp_pubsub_impl_consumer.erl

```diff
@@ -0,0 +1,200 @@
+    on_start/2,
+    on_stop/2,
+    on_get_status/2
+]).
+
+-include_lib("emqx/include/logger.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+-include_lib("emqx_resource/include/emqx_resource.hrl").
+
+-type mqtt_config() :: #{
+    mqtt_topic := emqx_types:topic(),
+    qos := emqx_types:qos(),
+    payload_template := string()
+}.
+-type config() :: #{
+    connect_timeout := emqx_schema:duration_ms(),
+    max_retries := non_neg_integer(),
+    pool_size := non_neg_integer(),
+    resource_opts := #{request_ttl := infinity | emqx_schema:duration_ms(), any() => term()},
+    service_account_json := emqx_bridge_gcp_pubsub_connector:service_account_json(),
+    any() => term()
+}.
+-type state() :: #{
+    connector_state := emqx_bridge_gcp_pubsub_connector:state()
+}.
+
+-export_type([mqtt_config/0]).
+
+-define(AUTO_RECONNECT_S, 2).
+
+%%-------------------------------------------------------------------------------------------------
+%% `emqx_resource' API
+%%-------------------------------------------------------------------------------------------------
+
+-spec callback_mode() -> callback_mode().
+callback_mode() -> async_if_possible.
+
+-spec query_mode(any()) -> query_mode().
+query_mode(_Config) -> no_queries.
+
+-spec on_start(resource_id(), config()) -> {ok, state()} | {error, term()}.
+on_start(InstanceId, Config) ->
+    case emqx_bridge_gcp_pubsub_connector:on_start(InstanceId, Config) of
+        {ok, ConnectorState} ->
+            start_consumers(InstanceId, ConnectorState, Config);
+        Error ->
+            Error
+    end.
+
+-spec on_stop(resource_id(), state()) -> ok | {error, term()}.
+on_stop(InstanceId, #{connector_state := ConnectorState}) ->
+    ok = stop_consumers(InstanceId),
+    emqx_bridge_gcp_pubsub_connector:on_stop(InstanceId, ConnectorState);
+on_stop(InstanceId, undefined = _State) ->
+    ok = stop_consumers(InstanceId),
+    emqx_bridge_gcp_pubsub_connector:on_stop(InstanceId, undefined).
+
+-spec on_get_status(resource_id(), state()) -> connected | disconnected.
+on_get_status(InstanceId, _State) ->
+    case
+        emqx_resource_pool:health_check_workers(
+            InstanceId,
+            fun emqx_bridge_gcp_pubsub_consumer_worker:health_check/1
+        )
+    of
+        true -> connected;
+        false -> connecting
+    end.
+
+%%-------------------------------------------------------------------------------------------------
+%% Internal fns
+%%-------------------------------------------------------------------------------------------------
+
+start_consumers(InstanceId, ConnectorState, Config) ->
+    #{
+        bridge_name := BridgeName,
+        consumer := ConsumerConfig0,
+        hookpoint := Hookpoint
+    } = Config,
+    ConsumerConfig1 = maps:update_with(topic_mapping, fun convert_topic_mapping/1, ConsumerConfig0),
+    TopicMapping = maps:get(topic_mapping, ConsumerConfig1),
+    PullWorkerMultiplier = maps:get(pull_worker_multiplier, ConsumerConfig1),
```

Comment:
Nit: m.b. `pull_worker_multiplier ` -> `consumer_workers_per_topic`? The name was a bit unclear until this point.

Reply:
:heavy_check_mark: